indico/migrations/versions/20181213_1110_cbe630695800_add_room_principals_table.py

```diff
@@ -0,0 +1,216 @@
+"""Add room principals table
+
+Revision ID: cbe630695800
+Revises: 252c0015c9a0
+Create Date: 2018-12-13 11:10:12.684382
+"""
+
+import json
+
+import sqlalchemy as sa
+from alembic import context, op
+from sqlalchemy.dialects import postgresql
+
+from indico.core.auth import multipass
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.core.db.sqlalchemy.protection import ProtectionMode
+
+
+# revision identifiers, used by Alembic.
+revision = 'cbe630695800'
+down_revision = '252c0015c9a0'
+branch_labels = None
+depends_on = None
+
+
+def _create_acl_entry(conn, room_id, user_id=None, group_id=None, mp_group_provider=None, mp_group_name=None,
+                      full_access=False, permissions=()):
+    permissions = list(permissions)
```

Comment:
Why not pass the list directly as an argument? It is being used only in one case with one element in a set.

Reply:
i don't like mutable default arguments even if it'd be safe in this case